Expose the inspected web view from _WKInspectorWindow

davidquesada opened the pull request:
#### 5bb44c8b1c3b4f3d3c289625412e57a0f6037f50
<pre>
Expose the inspected web view from _WKInspectorWindow
<a href="https://bugs.webkit.org/show_bug.cgi?id=251140">https://bugs.webkit.org/show_bug.cgi?id=251140</a>
rdar://104628163

Reviewed by Patrick Angle.

Introduce _WKInspectorWindow.inspectedWebView to read the web view inspected by the
inspector hosted in a _WKInspectorWindow.

* Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.mm:
    Let the compiler automatically synthesize the properties. Manually implementing them
    doesn&apos;t provide any benefit.
(-[_WKInspectorWindow isForRemoteTarget]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindowInternal.h: Copied from Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.h.
    As minor cleanup, add an internal header to redeclare the SPI as readwrite instead
    of exposing the properties as writable to WebKit clients who import the main header.
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.h:
    Add an optional parameter to createFrontendWindow() to specify a WebPageProxy being
    inspected. This will be used to populate the new `inspectedWebView` property.
(WebKit::WebInspectorUIProxy::inspectorWindow const): Deleted.
    Remove this method. It wasn&apos;t being used anywhere.
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
(WebKit::WebInspectorUIProxy::createFrontendWindow):
    If an inspected WebPageProxy was provided, look up the WKWebView to assign to
    _WKInspectorWindow.inspectedWebView.
(WebKit::WebInspectorUIProxy::platformCreateFrontendWindow):
    Pass the inspected page to createFrontendWindow().
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
    Add reference to _WKInspectorWindowInternal.h.

Canonical link: <a href="https://commits.webkit.org/259393@main">https://commits.webkit.org/259393@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2b105df938ce9896326c0aa182f8c5647b36525b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104759 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13857 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37675 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114034 "Built successfully") | 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108674 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14967 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4766 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97098 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/112958 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110519 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11573 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94595 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/108222 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93433 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26208 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/80772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7191 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27566 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/92649 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7292 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4149 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30206 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13351 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47118 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/101338 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6480 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9078 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/25278 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

patrickangle commented on Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.h:
```diff
@@ -27,11 +27,18 @@
 
-@property (nonatomic, getter=isForRemoteTarget) BOOL forRemoteTarget;
+@property (nonatomic, readonly, getter=isForRemoteTarget) BOOL forRemoteTarget;
+@property (nonatomic, nullable, readonly, weak) WKWebView *inspectedWebView;
```
This new property should probably get availability annotations for TBA versions. e.g. `@property (nonatomic, nullable, readonly, weak) WKWebView *inspectedWebView WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA))`

davidquesada replied:
I think I would just add `WK_API_AVAILABLE(macos(WK_MAC_TBA))`. This class is excluded on everything but non-Catalyst Mac (since NSWindow isn't available), so the iOS availability isn't particularly meaningful.

patrickangle replied:
Oh good point, yeah this is Mac only, so no need for iOS availability 😅

patrickangle commented on Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.mm:
```diff
@@ -24,16 +24,14 @@
-    return _forRemoteTarget;
-}
+@synthesize forRemoteTarget = _forRemoteTarget;
+@synthesize inspectedWebView = _inspectedWebView;
```
Are these properties not being automatically synthesized?

davidquesada replied:
I'm not sure. I'll remove the manual synthesis if that compiles. I was probably thinking (wrongly?) that auto synthesis isn't available when you're not using ARC.

patrickangle commented on Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.h:
```diff
@@ -140,7 +141,6 @@ class WebInspectorUIProxy
     void inspectedViewFrameDidChange(CGFloat = 0);
     void windowFrameDidChange();
     void windowFullScreenDidChange();
-    NSWindow *inspectorWindow() const { return m_inspectorWindow.get(); }
```
lol

patrickangle commented on Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
```diff
@@ -476,6 +476,9 @@ - (IBAction)_popUpButtonAction:(id)sender
 
+    if (inspectedPage())
+        [m_inspectorWindow setInspectedWebView:inspectedPage()->cocoaView().get()];
+
```
Should we put this inside `WebInspectorUIProxy::createFrontendWindow` instead so we do it at the same time we set `forRemoteTarget` along with setting up the rest of the window itself? It should effectively be a no-op for the `RemoteWebInspectorUIProxyMac` since we won't have an inspectedPage.

davidquesada replied:
Sure.

davidquesada committed:
Expose the inspected web view from _WKInspectorWindow
https://bugs.webkit.org/show_bug.cgi?id=251140
rdar://104628163

Reviewed by Patrick Angle.

Introduce _WKInspectorWindow.inspectedWebView to read the web view inspected by the
inspector hosted in a _WKInspectorWindow.

* Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.mm:
    Let the compiler automatically synthesize the properties. Manually implementing them
    doesn't provide any benefit.
(-[_WKInspectorWindow isForRemoteTarget]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindowInternal.h: Copied from Source/WebKit/UIProcess/API/Cocoa/_WKInspectorWindow.h.
    As minor cleanup, add an internal header to redeclare the SPI as readwrite instead
    of exposing the properties as writable to WebKit clients who import the main header.
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.h:
    Add an optional parameter to createFrontendWindow() to specify a WebPageProxy being
    inspected. This will be used to populate the new `inspectedWebView` property.
(WebKit::WebInspectorUIProxy::inspectorWindow const): Deleted.
    Remove this method. It wasn't being used anywhere.
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
(WebKit::WebInspectorUIProxy::createFrontendWindow):
    If an inspected WebPageProxy was provided, look up the WKWebView to assign to
    _WKInspectorWindow.inspectedWebView.
(WebKit::WebInspectorUIProxy::platformCreateFrontendWindow):
    Pass the inspected page to createFrontendWindow().
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
    Add reference to _WKInspectorWindowInternal.h.

Canonical link: https://commits.webkit.org/259393@main
